src/pages/index.js

```diff
@@ -41,62 +182,136 @@ editProfileMOB.addEventListener("click", () => {
+  maxImagePopup.open(cardData);
+};
+
 //***************************************************************************************************************************//
 //                                             Event Handlers                                                                //
 //***************************************************************************************************************************//
 
+//Handles profile edit submit
 function handleEditProfile(userData) {
-  userInfo.setUserInfo(userData);
-  editProfilePopup.popupForm.reset();
-  editProfileFormValidator.toggleButtonState();
-  editProfilePopup.close();
+  const submitButton = editProfileModal.querySelector(
+    "#edit-profile-modal-submit-button"
+  );
+  submitButton.textContent = "Saving...";
+
+  api
+    .patchProfileInfo(userData.name, userData.job)
+    .then((res) => {
+      userInfo.setUserInfo(res);
+      editProfilePopup.popupForm.reset();
+      editProfileFormValidator.toggleButtonState();
+      editProfilePopup.close();
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      submitButton.textContent = "Save";
+    });
 }
 
+//Handles new card submit
 function handleAddCardSubmit(cardData) {
-  createCard(cardData, "prepend");
-  addPlacePopup.popupForm.reset();
-  addPlaceFormValidator.toggleButtonState();
-  addPlacePopup.close();
+  const submitButton = addCardModal.querySelector(
+    "#add-card-modal-submit-button"
+  );
+  submitButton.textContent = "Saving...";
+
+  api
+    .addNewCard(cardData)
+    .then((res) => {
+      addNewCard(res, "prepend");
+      addPlacePopup.popupForm.reset();
+      addPlaceFormValidator.toggleButtonState();
+      addPlacePopup.close();
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      submitButton.textContent = "Create";
+    });
 }
 
-const handleImageClick = (cardData) => {
-  maxImagePopup.open(cardData);
-};
+//Handles card deletion
+let cardtoDeleteData;
 
-//***************************************************************************************************************************//
-//                                                     Objects                                                               //
-//***************************************************************************************************************************//
+//Handles confirm delete and sets delete data
+function handleConfirmDelete(data) {
+  confirmDeletePopup.open();
+  cardtoDeleteData = data;
+}
 
-const cardSection = new Section(initialCards, createCard, "#cards-list");
-const maxImagePopup = new PopupWithImage("#max-image-modal");
-const editProfilePopup = new PopupWithForm(
-  "#edit-profile-modal",
-  handleEditProfile
-);
-const addPlacePopup = new PopupWithForm("#add-card-modal", handleAddCardSubmit);
+//Deletes card
+async function handleDeleteCard() {
+  try {
+    api.deleteCard(cardtoDeleteData._id);
```

Comment:
Please, add `await` to wait for the server response otherwise the card is deleted immediately